component/otelcol/processor/processor.go

```diff
@@ -161,25 +161,34 @@ func (p *Processor) Update(args component.Arguments) error {
 	// supported telemetry signals.
 	var components []otelcomponent.Component
 
-	tracesProcessor, err := p.factory.CreateTracesProcessor(p.ctx, settings, processorConfig, nextTraces)
-	if err != nil && !errors.Is(err, otelcomponent.ErrDataTypeIsNotSupported) {
-		return err
-	} else if tracesProcessor != nil {
-		components = append(components, tracesProcessor)
+	var tracesProcessor otelcomponent.TracesProcessor
+	if len(next.Traces) > 0 {
```

Comment:
I am assuming that the intent is if next component cannot handle a specific signal type then it will not be set?